crates/biome_js_syntax/src/file_source.rs

```diff
@@ -280,6 +280,48 @@ impl JsFileSource {
             }
         }
     }
+
+    pub fn try_from_extension(extension: &str) -> Result<Self, FileSourceError> {
+        match extension {
+            "js" | "mjs" | "jsx" => Ok(Self::jsx()),
+            "cjs" => Ok(Self::js_script()),
+            "ts" => Ok(Self::ts()),
+            "mts" | "cts" => Ok(Self::ts_restricted()),
+            "tsx" => Ok(Self::tsx()),
+            "d.ts" | "d.mts" | "d.cts" => Ok(Self::d_ts()),
+            // TODO: Remove once we have full support of astro files
+            "astro" => Ok(Self::astro()),
+            // TODO: Remove once we have full support of vue files
+            "vue" => Ok(Self::vue()),
+            // TODO: Remove once we have full support of svelte files
+            "svelte" => Ok(Self::svelte()),
+            _ => Err(FileSourceError::UnknownExtension(
+                Default::default(),
+                extension.into(),
+            )),
+        }
+    }
+
+    pub fn try_from_well_known(file_name: &str) -> Result<Self, FileSourceError> {
+        // TODO: to be implemented
+        Err(FileSourceError::UnknownFileName(file_name.into()))
+    }
+
+    pub fn try_from_language_id(language_id: &str) -> Result<Self, FileSourceError> {
+        match language_id {
+            "javascript" => Ok(Self::js_module()),
+            "typescript" => Ok(Self::ts()),
+            "javascriptreact" => Ok(Self::jsx()),
+            "typescriptreact" => Ok(Self::tsx()),
```

Comment:
The language id comes from the VSCode language table. There's a link somewhere as a reference. Would it be possible to not lose the link, add it to this function (every language id) and document the meaning of the function?

Reply:
I added comments to each specific language file source. Language identifiers are also listed in the LSP spec, so I added that link too. And some language identifiers are registered by the extension (e.g. astro), that is also added in the comments.

Reply:
Can I merge this now? @ematipico